strawberry/schema/schema_converter.py

```diff
@@ -352,33 +378,35 @@ def _resolver(_source: Any, info: GraphQLResolveInfo, **kwargs):
             strawberry_info = _strawberry_info_from_graphql(info)
             _check_permissions(_source, strawberry_info, kwargs)
 
-            args, kwargs = _get_arguments(
+            field_args, field_kwargs = _get_arguments(
                 source=_source, info=strawberry_info, kwargs=kwargs
             )
 
-            result = field.get_result(
-                _source, info=strawberry_info, args=args, kwargs=kwargs
+            return field.get_result(
+                _source, info=strawberry_info, args=field_args, kwargs=field_kwargs
             )
 
-            if isasyncgen(result):
-
-                async def yield_results(results):
-                    async for value in results:
-                        yield value
-
-                return yield_results(result)
-
-            if iscoroutine(result):  # pragma: no cover
+        async def _async_resolver(_source: Any, info: GraphQLResolveInfo, **kwargs):
+            strawberry_info = _strawberry_info_from_graphql(info)
+            await _check_permissions_async(_source, strawberry_info, kwargs)
 
-                async def await_result(result):
-                    return await result
+            field_args, field_kwargs = _get_arguments(
+                source=_source, info=strawberry_info, kwargs=kwargs
+            )
 
-                return await_result(result)
+            return field.get_result(
+                _source, info=strawberry_info, args=field_args, kwargs=field_kwargs
+            )
 
-            return result
+        has_async_base_resolver = inspect.iscoroutinefunction(field.base_resolver)
```

Comment:
Hoisting this the field definition would also be good.

Reply:
Inspecting the base_resolver turned out to be unnecessary.